Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -286,4 +293,30 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
             appExtensionsBuildPhase.files?.append(pbxBuildFile)
         }
     }
+
+    func generateEmbedWatchBuildPhase(path: AbsolutePath,
+                                      target: Target,
+                                      graph: Graphing,
+                                      pbxTarget: PBXTarget,
+                                      fileElements: ProjectFileElements,
+                                      pbxproj: PBXProj) throws {
+        let targetDependencies = graph.targetDependencies(path: path, name: target.name)
+        let watchApps = targetDependencies.filter { $0.target.product == .watch2App }
```

Comment:
Interesting, I would have expected the watch app to have the iOS as a dependency, like we do with tests having the app as a dependency.

Reply:
Indeed this is an interesting one, using dependency paradigms for those can be thought of going either way depending how you look at it.

I opted to keep it consistent with how extensions are currently defined, they are fairly similar conceptually where they are companion products bundled with the main application - as such the application isn't complete without them.

```swift
 Target(name: "App",
        platform: .iOS,
        product: .app,
        bundleId: "io.tuist.App",
        infoPlist: "Support/App-Info.plist",
        sources: ["Sources/**"],
        dependencies: [

            // Extensions + Additional Bundled Products
            .target(name: "TodayExtension"),
            .target(name: "NotificationServiceExtension"),
            .target(name: "WatchApp"),
        ]),
```

For tests, its slightly different as often the test will actually depend on the app (`@testable import App`) additionally the test targets need to reference the host / target application either in their build settings or in their target attributes thus the dependency was specified in that direction.

Reply:
> I opted to keep it consistent with how extensions are currently defined, they are fairly similar conceptually where they are companion products bundled with the main application - as such the application isn't complete without them.

That's a fair reason 👍 and the example that you showed is very clear.